docs/design/proposals/adoption/resource-adoption.md

```diff
@@ -0,0 +1,225 @@
+
+1. The provided target ACK type does not exist within the cluster
+2. The target resource would override an existing resource
+3. The target resource type cannot be adopted
+
+## Adopted K8s Resource Tagging
+
+All adopted resources will be annotated with an ACK-specific annotation that differentiates it from a natively created resource. This annotation serves to ensure clarity for users about possible specification differences introduced during the adoption process, or for filtering by adopted status. However, once adopted into a cluster, a resource will be reconciled with the same logic as any natively created resource. 
+
+The proposed k8s adopted tag is `services.k8s.aws/adopted: true`
+
+The adopted annotation would remain on the custom resource for the remainder of its lifecycle, and should not be modified by any other manager operation. 
+
+## AWS Resource Tagging
+
+AWS resources adopted by an ACK controller will have tags retroactively applied (whenever possible) to indicate its association with an ACK deployment. The details of these tags are discussed in another issue ([#148](https://github.com/aws/aws-controllers-k8s/issues/148)). ACK managers should not place any conditions on the existence of ACK tags on the AWS resources to gate the adoption process. The manager should should override any existing ACK tags.
+
+## Resource Type Blocklisting
+
+Adoption, for some resources, is not possible due to any number of the following reasons:
+
+* They are ephemeral resources which don’t support additional lifecycle operations (one-time jobs)
+* The service cannot provide adequate information to produce a full specification
+* Unforeseen limitations within ACK or the service directly
+
+For any of these cases, the resource can be marked as “not adoptable” and the validation webhook will subsequently reject any adopted resources of this type. Therefore, the manager will never attempt to run the adoption process. This flag can be set in the generator configuration file with the `cannot_adopt` boolean option supplied for any resource.
+
+## Adoption Process Diagram
+
+![sequence-diagram](./images/adoption-sequence-diagram.png)
+# Potential Alternatives
+
+## Resource Adoption Annotation
+
+The existing custom resource definitions (every service resource) would accept an additional annotation that the manager interprets as an indication that the resource specification should be ignored and instead that it should be formed through adoption. For any resource marked with this annotation, upon creation, the manager would attempt to describe the AWS resource associated with the annotated value. If none such resource exists, the manager would update the state of the custom resource into a `AdoptionFailed` terminal state. If the resource does exist, the manager would continue the follow the same adoption process as defined in the [Proposed Implementation](#proposed-implementation) section.
```

Comment:
This harkens back to my comment above in The Problem section to describe the difficulty right now with this annotation and needing to fill in required fields in the Spec when the annotation essentially signals to the service controller to ignore the values of those Spec fields. :)

Reply:
I'd almost prefer that we remove this `services.k8s.aws/arn` annotation from ACK entirely and have a single way of signaling resource adoption (using AdoptedResource CRs)

Reply:
I think this section is worth removing. I have listed it under "Potential Alternatives" but realistically I don't think we will ever go down this route because of the problems we have both listed. Plus it's not super clear in this format that this is not part of the proposed solution.